R/tests/testthat/test_generate_strings.R

```diff
@@ -0,0 +1,175 @@
+# R. Samman
+# November 07 2024
+
+rm(list = ls())
+
+if (!require(testthat)) install.packages('testthat')
+library(testthat)
+
+source("/Users/raneemsamman/Documents/GitHub/paRsynth/R/generate_strings.R")
+
+# Checking that the number of groups and individuals is correct
+
+    # what do we want to do?
+        # 1. check n_groups
+        # 2. check n_individuals (per group?)
+        # 3. n_calls (i am adding this here bc it will be a double test for n_individuals)
+    # how are we doing that?
+        # 1. set a hypothetical set of parameters
+        # 2. call the function using my hypothetical parameters
+        # 3. check if the number of rows created is correct:
+            # n_groups * n_individuals * n_calls should give me the total number of rows
+        # 4. check if n_groups created is correct:
+            # use equal for the number of gruops (unique ones so that it doesnt double count) in the group column
+        # 5. check if n_individuals created is correct
+            # use equal for the number of individuals (unique ones) in the individuals column
+
+test_that("The function generates the number of groups and individuals", {
+
+# Avoid library calls and other changes to the virtual environment
+  # See https://r-pkgs.org/testing-design.html
+  withr::local_package("tidyverse")
+  withr::local_package("plyr")
+  withr::local_package("dplyr")
+  withr::local_package("lubridate")
+  withr::local_package("data.table")
+  
+  # Just for code development
+#   library(tidyverse)
+#   library(lubridate)
+#   library(testthat)
+#   library(data.table)
+#   library(stringer)
+
+#   # Create a temporary directory for testing. Files will be written and read here
+#   path <- "/Users/raneemsamman/Desktop"
+#   data_dir <- "tmp_tests"
+#   tmp_path <- file.path(path, data_dir)
```

Comment:
You can remove these lines that are commented out for creating the temporary directory